### PR TITLE
Adjusts the default color for search border

### LINKED ISSUE #27004: Search and replace - current-match is hard to see
**Godot version:**
3.1

**OS/device including version:**
Not relevant

**Issue description:**
Search & replace - current-match is hard to see. 
As the editor marks all the occurrences of the search, the current match looks the same as all the other occurrences.
Need to do something to make it stand forward, especially when there are couple of matches on the same line. Maybe give a different color.

**Steps to reproduce:**

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by siddharth952**
Fixes #27004 
Changed default current line colour to yellow, such that user is able to distinguish between the lines when using Find And Replace.

### CODE REVIEW

**Comment by volzhs** [comment]
does this make current line noticeable with **Light theme** also?

**Comment by siddharth952** [comment]
@volzhs Yes.

**Comment by groud** [comment]
Could you provide a screenshot please ?

**Comment by piyushdagar** [comment]
It will not work if the search results are in the same line. You should have also changed the search_result_border_color.

**Comment by siddharth952** [comment]
@piyushdagar  Yeah, thats right cause this causes current line colour to be yellow while editing the text.

Making changes to search_result_border_color

**Comment by piyushdagar** [comment]
Also reduce alpha of search_result_color (to maybe 0.3)

**Comment by siddharth952** [comment]
Screenshot 2019-03-15 at 11 14 58 PM

File: doc/classes/TextEdit.xml [hunk 1]
```diff
@@ -401,7 +401,7 @@
 			String value of the [TextEdit].
 		</member>
 		<member name="v_scroll_speed" type="float" setter="set_v_scroll_speed" getter="get_v_scroll_speed">
-			If [code]true[/code], enables text wrapping when it goes beyond he edge of what is visible.
+			Adjust [code]Vertical[/code] Scroll Sensitivity of Text Editor.
```
Comment (bojidar-bg) [inline doc/classes/TextEdit.xml]: Move the old documentation to `wrap_enabled`.

File: editor/editor_settings.cpp [hunk 2]
```diff
@@ -647,7 +647,7 @@ void EditorSettings::_load_default_text_editor_theme() {
 	_initial_set("text_editor/highlighting/breakpoint_color", Color(0.8, 0.8, 0.4, 0.2));
 	_initial_set("text_editor/highlighting/code_folding_color", Color(0.8, 0.8, 0.8, 0.8));
 	_initial_set("text_editor/highlighting/search_result_color", Color(0.05, 0.25, 0.05, 1));
-	_initial_set("text_editor/highlighting/search_result_border_color", Color(0.1, 0.45, 0.1, 1));
+	_initial_set("text_editor/highlighting/search_result_border_color", Color(255, 255, 0, 0.5));
```
Comment (bojidar-bg) [inline editor/editor_settings.cpp]: Color() uses numbers in the range 0..1, not 0..255.

File: doc/classes/TextEdit.xml [hunk 3]
```diff
@@ -404,6 +404,7 @@
 			Adjust [code]Vertical[/code] Scroll Sensitivity of Text Editor.
 		</member>
 		<member name="wrap_enabled" type="bool" setter="set_wrap_enabled" getter="is_wrap_enabled">
+            If [code]true[/code], enables text wrapping when it goes beyond the edge of what is visible.
```
Comment (akien-mga) [inline doc/classes/TextEdit.xml]: Wrong indentation, should use tabs.

File: doc/classes/TextEdit.xml [hunk 4]
```diff
@@ -404,6 +404,7 @@
 			Adjust [code]Vertical[/code] Scroll Sensitivity of Text Editor.
```
Comment (akien-mga) [inline doc/classes/TextEdit.xml]: There's no reason to use Title Capitalization here. Also no reason to make "Vertical" inline code as it's not a code identifier. "Vertical scroll sensitivity." would be fine.

**Comment by akien-mga** [comment]
Please squash the three commits together and fix the pending comments.
For future PRs, please make one specific PR for each change, default color of the search border and TextEdit documentation fixes are not related.

**Comment by siddharth952** [comment]
@akien-mga Yes I did try, there seems to be a problem with the branch I'll sort it out within 1-2 days

**Comment by siddharth952** [comment]
@akien-mga Should I close this pull and request two new pull requests? I have some problem with my fork I was working with the Collaboration feature for GSoC and by mistake it was reflected on this branch.

Sorry for the inconvenience caused will be avoided on future PRs.

**Comment by akien-mga** [comment]
> Should I close this pull and request two new pull requests? I have some problem with my fork I was working with the Collaboration feature for GSoC and by mistake it was reflected on this branch.

If you want yes. Otherwise you can force push an update to this PR's branch to make it have only one relevant commit, and open another PR for the other change.

**Comment by siddharth952** [comment]
@akien-mga I have updated the PR and made a separate PR for the other issue.

**Review by Calinou — CHANGES_REQUESTED** [review]
I suggest using `Color(0.41, 0.61, 0.91, 0.38)` for a more subtle appearance of non-selected matches:

search_border_dark

search_border_light

Also, this only changes the default color for the **Default** text editor theme, not the **Adaptive** one. The same color should be applied in both themes, so please copy the color there too: https://github.com/godotengine/godot/blob/7de83375eaab725e064aceed9a803f3826cd870b/editor/editor_themes.cpp#L1155

**Comment by siddharth952** [comment]
@Calinou 
We are modifying the value of text_editor/highlighting/search_result_border_color to be (0.41, 0.61, 0.91, 0.38) so if we use,

set_initial_value("text_editor/highlighting/search_result_border_color", search_result_border_color, true)

in Adaptive will it not copy the color values we set in _initial_set?

**Comment by Calinou** [comment]
@siddharth952 The `_initial_set` only affects the Default theme, not the Adaptive theme.

**Comment by siddharth952** [comment]
So to set the color value for the Adaptive theme we replace with the following ?

set_initial_value("text_editor/highlighting/search_result_border_color",Color(0.41, 0.61, 0.91, 0.38) , true);

**Comment by siddharth952** [comment]
> So to set the color value for the Adaptive theme we replace with the following ?
> 
> set_initial_value("text_editor/highlighting/search_result_border_color",Color(0.41, 0.61, 0.91, 0.38) , true);

@Calinou

**Comment by Calinou** [comment]
@siddharth952 Yes, make sure to remove the variable above which will become unused (and add a space after the `,` to respect the established code style).

**Review by Calinou — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Merged. Thanks for your contribution, and thanks for the effort in incorporating our feedback!

**Pull request merged**